python/cudf/cudf/_lib/nvtx/utils/cached.py

```diff
@@ -0,0 +1,11 @@
+class CachedInstanceMeta(type):
```

Comment:
Is this class vendored from somewhere? If so we should add a comment indicating where.

Reply:
Not really. It's based on a pattern described in the Python Cookbook